cg/meta/store/mip_rna.py

```diff
@@ -1,102 +1,133 @@
 """Builds MIP RNA bundle for linking in Housekeeper"""
+import logging
+from pathlib import Path
 
+import ruamel.yaml
 
-def build_bundle_rna(config_data: dict, sampleinfo_data: dict) -> dict:
+from cg.exc import AnalysisNotFinishedError
+from cg.meta.store.base import (
+    get_case,
+    reset_case_action,
+    add_new_analysis,
+    include_files_in_housekeeper,
+    add_bundle,
+)
+
+LOG = logging.getLogger(__name__)
+
+
+def gather_files_and_bundle_in_housekeeper(config_stream, hk_api, status):
+    """Function to gather files and bundle in housekeeper"""
+    bundle_data = add_analysis(config_stream)
+
+    results = add_bundle(hk_api, bundle_data)
+    bundle_obj, version_obj = results
+
+    case_obj = get_case(bundle_obj, status)
+    reset_case_action(case_obj)
+    new_analysis = add_new_analysis(bundle_data, case_obj, status, version_obj)
+    version_date = version_obj.created_at.date()
+    LOG.info("new bundle added: %s, version %s", bundle_obj.name, version_date)
+    include_files_in_housekeeper(bundle_obj, hk_api, version_obj)
+
+    return new_analysis
+
+
+def add_analysis(config_stream):
+    """Gather information from MIP analysis to store."""
+    config_raw = ruamel.yaml.safe_load(config_stream)
+    config_data = parse_config(config_raw)
+    sampleinfo_raw = ruamel.yaml.safe_load(Path(config_data["sampleinfo_path"]).open())
+    sampleinfo_data = parse_sampleinfo(sampleinfo_raw)
+
+    if sampleinfo_data["is_finished"] is False:
+        raise AnalysisNotFinishedError("analysis not finished")
+
+    deliverables_raw = ruamel.yaml.safe_load(Path(config_raw["store_file"]).open())
+    new_bundle = build_bundle(config_data, sampleinfo_data, deliverables_raw)
+
+    return new_bundle
+
+
+def build_bundle(config_data: dict, sampleinfo_data: dict, deliverables: dict) -> dict:
     """Create a new bundle for RNA."""
     data = {
-        'name': config_data['case'],
-        'created': sampleinfo_data['date'],
-        'pipeline_version': sampleinfo_data['version'],
-        'files': get_files_rna(config_data, sampleinfo_data),
+        "name": config_data["case"],
+        "created": sampleinfo_data["date"],
+        "pipeline_version": sampleinfo_data["version"],
+        "files": get_files(deliverables),
     }
     return data
 
 
-def get_files_rna(config_data: dict, sampleinfo_data: dict) -> dict:
+def get_files(deliverables: dict) -> dict:
```

Comment:
I think this one should be in store/base.py for easy reuse by other pipelines. This should be agnostic to all workflows using the deliverables files right?

Reply:
Moved to `base.py` and made pipeline agnostic.